plugin.go

```diff
@@ -38,30 +50,206 @@ func (p *Plugin) CobraAddCommands(rootCmd *cobra.Command) error {
 	var port string
 	var proxyClient string
 	var useSwaggerUI bool
+	var foreground bool
+	var all bool
 	var cmd = &cobra.Command{
-		Use:   "web",
-		Short: "Starts web server",
+		Use:       "web",
+		Short:     "Starts web server",
+		Args:      cobra.MatchAll(cobra.RangeArgs(0, 1), cobra.OnlyValidArgs),
+		ValidArgs: []string{stopArg},
 		RunE: func(cmd *cobra.Command, args []string) error {
 			// Don't show usage help on a runtime error.
 			cmd.SilenceUsage = true
 
+			// If 'stop' arg passed, try to kill process and remove PID file.
+			if len(args) > 0 && args[0] == stopArg {
+				return stopServer(all)
+			}
+
 			runOpts := &server.RunOptions{
 				Addr:        fmt.Sprintf(":%s", port), // @todo use proper addr
 				APIPrefix:   APIPrefix,
 				SwaggerJSON: useSwaggerUI,
 				ProxyClient: proxyClient,
 				// @todo use embed fs for client or provide path ?
 			}
-
 			prepareRunOption(p, runOpts)
 
+			if !isPortFree(port) {
+				return fmt.Errorf("web server port %s you are trying to use is not available", port)
+			}
+
+			if foreground {
+				return server.Run(cmd.Context(), p.app, runOpts)
+			}
+
+			if len(os.Getenv(cobraBackgroundEnvVar)) == 0 {
+				return runBackground(cmd, generatePidFile(port))
+			}
+
 			return server.Run(cmd.Context(), p.app, runOpts)
 		},
 	}
 	cmd.Flags().StringVarP(&port, "port", "p", "8080", `Web server port`)
 	cmd.Flags().BoolVarP(&useSwaggerUI, "swagger-ui", "", false, `Serve swagger.json on /api/swagger.json and Swagger UI on /api/swagger-ui`)
+	cmd.Flags().BoolVarP(&foreground, "foreground", "", false, `Run server as foreground process`)
+	cmd.Flags().BoolVarP(&all, "all", "", false, `Stop all background applications`)
```

Comment:
Is this option meant to stop webui from running ? If yes, probably need a more intuitive name than `all`

Also, does not seem to work:
```
➜  ski-platform git:(master) plasmactlui web
Web running in background with PID: 38049

➜  ski-platform git:(master) plasmactlui web --all
Error: web server port 8080 you are trying to use is not available
```